### ISSUE TITLE
RFE: Login for root when there is no entry in the authorization file

### ISSUE THREAD

**Issue by dlippold** [Yubico/pam-u2f#154]
When root wants to make 2-factor authorization obligatory for the users he/she can add the following line to the files in `/etc/pam.d/`;

> auth   required   pam_u2f.so cue

But when root forgot to register his/her own token before, i.e. storing the value of `pamu2fcfg` in the authorization file (e.g. `/root/.config/Yubico/u2f_keys`), or root made a mistake when registering the token root will not be able to login anymore.

root could add `nouserok` to the line above to avoid such a situation but then the usage of 2-factor authorization would not be obligatory anymore (e.g. a user could remove his/her file `~/.config/Yubico/u2f_keys`).

Therefore I want to suggest that without `nouserok` an entry in the authorization file should be obligatory for the users but should not be obligatory for root (i.e. root can login without the second factor when there is no entry in authorization file for root).

It may seem strage that a login without a second factor should be possible just for root but that is only the case when there is **no** entry in the authorization file for root. Normally root will have an entry in the authorization file and then this should still be checked of course (i.e. someone has to use the second factor to become root in that case).

**Comment by a-dma** [comment]
I understand the spirit behind this request, but I don't know if this should be treated in a special way. What happens if root forgets their password? There is no workaround for letting root in. What happens if root misplaces the SSH private key? There is no workaround for letting root in. Why should it be different if root incorrectly configures the module?

It can probably be a dedicated option so that each system owner can choose how to configure it, but it's yet-another-knob...

**Comment by dlippold** [comment]
Ok, I understand that.
 
Maybe at least a notice can be added to the documentation that root should check the function of its second factor immediately after making 2-factor authentication obligatory (when a root shell is still open) or root should check the function first with "nouserok".

**Comment by a-dma** [comment]
Absolutely. The README file needs some rewriting anyway. I'll leave this open until that task is completed.
